flow/raster_cache.h

```diff
@@ -149,12 +149,19 @@ class RasterCache {
       const SkMatrix& ctm,
       bool checkerboard) const;
 
-  static SkIRect GetDeviceBounds(const SkRect& rect, const SkMatrix& ctm) {
+  static SkRect GetDeviceBounds(const SkRect& rect, const SkMatrix& ctm) {
+#ifndef SUPPORT_FRACTIONAL_TRANSLATION
     SkRect device_rect;
     ctm.mapRect(&device_rect, rect);
     SkIRect bounds;
     device_rect.roundOut(&bounds);
-    return bounds;
+    return SkRect::MakeLTRB(bounds.fLeft, bounds.fTop, bounds.fRight,
+                            bounds.fBottom);
+#else
```

Comment:
These 2 branches share a few lines of code so the ifndef could be reduced a bit.

Even more because I believe that SkRect has a "round out to SkRect" method which can be executed on "itself" which means only 1 line of code needs to be inside the ifndef.

Reply:
But the math isn't really complete here.

When computing the size of the cache image we need something along the lines of roundOut, but it isn't being done. More comments in the Rasterize method...

Reply:
Perhaps this method should perform no roundOut in any case and let the calling method figure out the right way to handle non-integer bounds?